xrmocap/data/dataset/bottom_up_mview_mperson_dataset.py

```diff
@@ -0,0 +1,177 @@
+            cam_k_dim=cam_k_dim,
+            logger=logger)
+        
+        
+    def __getitem__(
+        self, index: int
+    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor,
+               torch.Tensor, bool, dict]:
+        """Get item from the dataset.
+
+        Args:
+            index (int): Index in dataset.
+
+        Returns:
+            mview_img_tensor (torch.Tensor):
+                Multi-view image tensor in shape (n_view, h, w, c).
+            k_tensor (torch.Tensor):
+                Multi-view intrinsic tensor in shape (n_view, 3, 3).
+            r_tensor (torch.Tensor):
+                Multi-view rotation mat tensor in shape (n_view, 3, 3).
+            t_tensor (torch.Tensor):
+                Multi-view translation vector tensor in shape (n_view, 3, 3).
+            kps3d (torch.Tensor):
+                Multi-view kps3d tensor in shape (n_person, n_kps, 4),
+                while the last dim is confidence. If kps3d[p, k, 4] == 0,
+                kps3d[p, k] is invalid and do not use the data.
+            end_of_clip (bool):
+                Whether it is the last frame of this clip. When shuffled,
+                it is always False.
+            kw_data (dict):
+                Dict for keyword data. bbox and kps2d can be found here.
+        """
+        if index >= len(self):
+            raise StopIteration
+        scene_idx, frame_idx, end_of_clip = self.process_index_mapping(index)
+        # load multi-view images
+        img_paths = self.image_list[scene_idx][frame_idx]
+        mview_img_list = []
+        for rela_path in img_paths:
+            img_path = os.path.join(self.data_root, rela_path)
+            img_tensor = self.img_pipeline(img_path)
+            mview_img_list.append(img_tensor)
+        mview_img_tensor = torch.stack(mview_img_list)
+        
+        k_list = []
+        r_list = []
+        t_list = []
+        # prepare multi-view cameras
+        for fisheye_param in self.fisheye_params[scene_idx]:
+            k_list.append(
+                torch.tensor(fisheye_param.get_intrinsic(self.cam_k_dim)))
+            r_list.append(torch.tensor(fisheye_param.get_extrinsic_r()))
+            t_list.append(torch.tensor(fisheye_param.get_extrinsic_t()))
+        k_tensor = torch.stack(k_list)
+        r_tensor = torch.stack(r_list)
+        t_tensor = torch.stack(t_list)
+        # prepare kps3d
+        keypoints3d = self.gt3d[scene_idx]
+        kps3d = keypoints3d.get_keypoints()[frame_idx]
+        # if this frame is the end of clip(scene)
+        end_of_clip = end_of_clip and not self.shuffled
+        # prepare keyword data
+        kw_data = {}
+
+        mview_keypoints2d_list = self.percep_keypoints2d[scene_idx]
+        mview_kps2d_list = []
+        n_view = mview_img_tensor.shape[0]
+        for view_idx in range(n_view):
+            mview_kps2d_list.append(mview_keypoints2d_list[view_idx][frame_idx])
+        kw_data = mview_kps2d_list
```

Comment:
`kw_data` is a dict, the key should be `joints` and `pafs` in this dataset?

Reply:
some confusing definitions exist. I will resolve it. `kw_data` is a `list` of `dict` with keys of `joints` and `pafs`.